ASM/src/big_poe.asm

```diff
@@ -0,0 +1,30 @@
+; Since we changed the ocflags collision to be able to interact with everything, check if the actor the big poe soul
+; has collided with either Link or Epona, to avoid weird interactions like throwing a bomb at the soul to collect it.
+big_poe_soul_collision:
+    andi    t1, t0, 0x0002             ; vanilla condition. if t1 = 0, then the soul won't interact with whatever was colliding.
+    beq     t1, $zero, @@no_collision
+    nop
+
+    lw      t2, 0x0240(s0)
+    lh      t3, 0x0000(t2)             ; t3 = this->collider.base.oc->id, so the id of the actor that collided with the soul
+
+    beqz    t3, @@collision            ; compare to 0, the id of Player actor
+    nop
+
+    addiu   t2, $zero, 0x0014          ; 0014 is the horse actor
+    bne     t2, t3, @@no_collision     ; if epona is colliding, check if Link is indeed riding
+    nop
+
+    lw      v0, 0x1C44(a1)             ; v0 = Player
```

Comment:
it doesn't seem like a1 should be assumed to be PlayState at this point. This works by chance because Actor_TalkOfferAccepted doesn't end up touching a1. A safer thing to do would be to get PlayState from 0x34(sp) where it is saved prior to calling Actor_TalkOfferAccepted. This would make sure this hack doesn't break if Actor_TalkOfferAccepted were ever changed in the future

Reply:
or at very least use a3 it seems like the function tries to make sure that PlayState is kept there